util/llbutil/authprovider/podman.go

```diff
@@ -0,0 +1,138 @@
+//go:generate hel --output helheim_mocks_test.go
+
+package authprovider
+
+import (
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/docker/cli/cli/config"
+	"github.com/docker/cli/cli/config/configfile"
+	"github.com/moby/buildkit/session"
+	"github.com/moby/buildkit/session/auth/authprovider"
+	"github.com/pkg/errors"
+)
+
+const (
+	dockerDockerhubKey = "https://index.docker.io/v1/"
+	podmanDockerhubKey = "docker.io"
+	podmanAuthFile     = "auth.json"
+)
+
+// OS contains methods that are similar to the os package functions. It is
+// provided so that os-level functions may be mocked.
+type OS interface {
+	Open(string) (io.ReadCloser, error)
+	Getenv(string) string
+}
+
+type defaultOS struct{}
+
+func (o defaultOS) Open(path string) (io.ReadCloser, error) {
+	return os.Open(path)
+}
+
+func (o defaultOS) Getenv(name string) string {
+	return os.Getenv(name)
+}
+
+type podmanCfg struct {
+	os OS
+}
+
+// PodmanOpt is an option which may be used when constructing a podman auth
+// provider.
+type PodmanOpt func(podmanCfg) podmanCfg
+
+// WithOS returns an option that provides custom OS-level functions for the
+// podman auth provider to use.
+func WithOS(o OS) PodmanOpt {
+	return func(c podmanCfg) podmanCfg {
+		c.os = o
+		return c
+	}
+}
+
+func NewPodman(stderr io.Writer, opts ...PodmanOpt) session.Attachable {
+	conf := podmanCfg{
+		os: defaultOS{},
+	}
+	for _, o := range opts {
+		conf = o(conf)
+	}
+	if authfile := conf.os.Getenv("REGISTRY_AUTH_FILE"); authfile != "" {
```

Comment:
nit: I usually find it cleaner to use os.Lookupenv since you can distinguish empty string vs not set (also I think the boolean check is a little nicer)

Reply:
This code can't use the empty string, since that could cause it to read files that it shouldn't. `os.LookupEnv` can return `"", true`, so using `LookupEnv` would look like this:

```
if authfile, ok := conf.os.LookupEnv("REGISTRY_AUTH_FILE"); ok && authfile != "" {
```

Since this code needs to treat an unset variable and a variable set to the empty string the same, it's more idiomatic to use `Getenv`.